service/src/main/java/org/jboss/sbomer/service/feature/sbom/errata/event/release/ReleaseTextOnlyAdvisoryEventsListener.java

```diff
@@ -0,0 +1,451 @@
+                        erratum.getDetails().get().getFulladvisory(),
+                        erratum.getDetails().get().getId());
+
+                // If the notes contains a "deliverables" field, search the latest successful generations triggered by
+                // the request event
+                V1Beta1RequestRecord advisoryManifestsRecord = sbomService
+                        .searchLastSuccessfulAdvisoryRequestRecord(requestEvent.getId(), config.getAdvisoryId());
+                manifestsPurls = advisoryManifestsRecord.manifests()
+                        .stream()
+                        .map(manifest -> manifest.rootPurl())
+                        .toList();
+            }
+
+            List<Sbom> sboms = findSbomsByPurls(manifestsPurls);
+            createReleaseManifestsForTextOnlyAdvisories(
+                    requestEvent,
+                    erratum,
+                    sboms,
+                    event.getReleaseGenerations(),
+                    toolVersion,
+                    productType);
+        } catch (Exception e) {
+            log.error(
+                    "An error occured during the creation of release manifests for event '{}'",
+                    requestEvent.getId(),
+                    e);
+            markRequestFailed(requestEvent, event.getReleaseGenerations().values());
+        }
+
+        // Let's trigger the update of statuses and advisory comments
+        doUpdateGenerationsStatus(event.getReleaseGenerations().values());
+    }
+
+    private List<Sbom> findSbomsByPurls(List<String> purls) {
+        if (purls == null || purls.isEmpty()) {
+            return Collections.emptyList();
+        }
+        return purls.stream().map(sbomService::findByPurl).filter(Objects::nonNull).toList();
+    }
+
+    @Transactional(value = TxType.REQUIRES_NEW)
+    protected void markRequestFailed(RequestEvent requestEvent, Collection<SbomGenerationRequest> releaseGenerations) {
+        String reason = "An error occured during the creation of the release manifest";
+        log.error(reason);
+
+        requestEvent = requestEventRepository.findById(requestEvent.getId());
+        requestEvent.setEventStatus(RequestEventStatus.FAILED);
+        requestEvent.setReason(reason);
+
+        for (SbomGenerationRequest generation : releaseGenerations) {
+            generation = generationRequestRepository.findById(generation.getId());
+            generation.setStatus(SbomGenerationStatus.FAILED);
+            generation.setReason(reason);
+        }
+    }
+
+    protected void createReleaseManifestsForTextOnlyAdvisories(
+            RequestEvent requestEvent,
+            Errata erratum,
+            List<Sbom> sboms,
+            Map<String, SbomGenerationRequest> releaseGenerations,
+            String toolVersion,
+            Component.Type productType) {
+
+        String productName = erratum.getDetails().get().getProduct().getName();
```

Comment:
Here, `sboms` could be empty, maybe it would be good to add an information about it and return early?